torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2037,49 +2587,86 @@ def create_symbolic_sizes_strides_storage_offset(
                 )
         assert all(x is not None for x in stride)
 
-        sym_sizes = [self.create_symintnode(i, hint=hint) for i, hint in zip(size, ex.size())]
+        sym_sizes = [
+            self.create_symintnode(sym, hint=hint, source=TensorPropertySource(source, TensorProperty.SIZE, i), positive=True)
+            for i, (sym, hint) in enumerate(zip(size, ex.size()))
+        ]
         sym_stride = []
         for i, stride_expr in enumerate(stride):
             # NB: Don't duck size the stride; instead use the expression
             # we computed
             assert stride_expr is not None
-            sym_stride.append(self.create_symintnode(stride_expr, hint=ex.stride(i)))
+            sym_stride.append(self.create_symintnode(
+                stride_expr, hint=ex.stride(i), source=TensorPropertySource(source, TensorProperty.STRIDE, i), positive=True
+            ))
         sym_storage_offset = self.create_symintnode(self.create_symbol(
             ex.storage_offset(),
             TensorPropertySource(source, TensorProperty.STORAGE_OFFSET),
             # TODO: This should be DYNAMIC, using DUCK for BC
             dynamic_dim=DimDynamic.DUCK,
             constraint_dim=None,
-        ), hint=ex.storage_offset())
+        ), hint=ex.storage_offset(), source=TensorPropertySource(source, TensorProperty.STORAGE_OFFSET))
         return sym_sizes, sym_stride, sym_storage_offset
 
     # If you know what the current hint value of the SymInt to be created
     # is, pass it into hint.  Otherwise, pass None and we will make our best
     # guess
-    def create_symintnode(self, sym: "sympy.Expr", *, hint: Optional[int]):
+    def create_symintnode(
+            self,
+            sym: "sympy.Expr",
+            *,
+            hint: Optional[int],
+            source: Optional[Source] = None,
+            positive: Optional[bool] = None
+    ):
+        if _translation_validator_enabled() and source is not None:
+            # Create a new symbol for this source.
```

Comment:
I can believe this is the best way to do it, but I'm a little fuzzy on the necessity of creating a new symbol and then asserting equality. Is this to prevent simplifications from happening to the input symbols?

Reply:
If so, it seems important to have a clear nomenclature that distinguishes these two cases, and talk through how these are treated differently in ShapeEnv

Reply:
As far as I understand, `sym` may not always be a `sympy.Symbol`. That's why we create a symbol for this source, and then assert equality.

Reply:
specifically, it may just be an int, is that right?